src/test/kotlin/se/uulm/snowballr/backend/service/project/GetDecisionStatisticsForStageTest.kt

```diff
@@ -0,0 +1,155 @@
+package se.uulm.snowballr.backend.service.project
+
+import io.mockk.coEvery
+import kotlinx.coroutines.test.runTest
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.assertDoesNotThrow
+import org.junit.jupiter.api.assertThrows
+import se.uulm.snowballr.backend.DataBuilder
+import se.uulm.snowballr.backend.TestSpecificException
+import se.uulm.snowballr.backend.model.SnowballRException
+import se.uulm.snowballr.backend.model.SnowballRException.UnauthorizedException
+import se.uulm.snowballr.backend.service.MainServiceTest
+import snowballr.ProjectOuterClass
+import snowballr.UserOuterClass
+import java.util.UUID
+import kotlin.test.assertEquals
+
+class GetDecisionStatisticsForStageTest : MainServiceTest() {
+    @Test
+    fun `When a server admin retrieves the decision statistics, then no exception is thrown`() = runTest {
+        val user = DataBuilder.createExampleUser(role = UserOuterClass.UserRole.USER_ROLE_ADMIN)
+        val project = DataBuilder.createExampleProject()
+
+        val request = ProjectOuterClass.Project.Information.DecisionStatistics.Get
+            .newBuilder()
+            .setProjectId(project.id.toString())
+            .setStage(0)
+            .build()
+
+        mockCurrentUser(user)
+        coEvery { projectMemberRepoMock.getProjectMembers(project.id) } returns emptyList()
+        coEvery { projectRepoMock.getProjectById(project.id) } returns Result.success(project)
+        coEvery { projectPaperRepoMock.getAllProjectPapersForProject(project.id) } returns emptyList()
+
+        assertDoesNotThrow { mainService.getDecisionStatisticsForStage(request) }
+    }
+
+    @Test
+    fun `When a project member retrieves the decision statistics, then no exception is thrown`() = runTest {
+        val user = DataBuilder.createExampleUser(role = UserOuterClass.UserRole.USER_ROLE_DEFAULT)
+        val project = DataBuilder.createExampleProject()
+        val projectMember = DataBuilder.createExampleProjectMember(userId = user.id, projectId = project.id)
+
+        val request = ProjectOuterClass.Project.Information.DecisionStatistics.Get
+            .newBuilder()
+            .setProjectId(project.id.toString())
+            .setStage(0)
+            .build()
+
+        mockCurrentUser(user)
+        coEvery { projectMemberRepoMock.getProjectMembers(project.id) } returns listOf(projectMember)
+        coEvery { projectRepoMock.getProjectById(project.id) } returns Result.success(project)
+        coEvery { projectPaperRepoMock.getAllProjectPapersForProject(project.id) } returns emptyList()
+
+        assertDoesNotThrow { mainService.getDecisionStatisticsForStage(request) }
+    }
+
+    @Test
+    fun `When a non-member retrieves the decision statistics, then an UnauthorizedException is thrown`() = runTest {
```

Comment:
```suggestion
    fun `When a non member retrieves the decision statistics, then an UnauthorizedException is thrown`() = runTest {
```
We always used it this way.

Reply:
In my opinion, both are correct, but since we used your variant more often in the project, I will include this for the sake of consistency.

Reply:
Ah ok, then I will not highlight this anymore if both are correct.